vignettes/delta-method.Rmd

```diff
@@ -119,3 +119,172 @@ Take the second derivative by subtracting each of the levels from the base level
+
+# Example
+
+Say we're fitting a logistic regression using the `margex` data, and we're interested in the predicted outcome for different treatment groups:
+
+```{r}
+library(modmarg)
+data(margex)
+lg <- glm(outcome ~ treatment * age, data = margex, family = 'binomial')
+summary(lg)
+```
+
+where the `treatment` variable $\tau \in {0, 1}$. What is the average outcome when $\tau = 0$ or $\tau = 1$?
+
+Based on these results alone, we don't really know what the treatment effect is, both because we can't directly interpret the coefficients, because there's an interaction term, and because the effect of each covariate depends on the levels of the other covariates.
+
+## Point estimates
+
+To get an estimate of the average levels of the outcome for different values of treatment, we can set treatment to 0 or 1 for the entire dataset, generate predicted outcomes for the dataset, and averaging the results (in other words, what would the outcome have been, if everybody was in the control / treatment group?). We'll use the `predict` and `plogis` functions to do this. Predict will calculate the linear predictors (in this case, setting $\text{treatment} = t$: 
+\[
+\hat{y}_i = \alpha + \beta_1 \cdot t + \beta_2 \cdot \text{age}_i + \beta_3 \cdot (t \cdot \text{age}_i)
+\]
+`plogis` is the inverse link function $f(z) = \frac{1}{1 + \exp(-z)}$, which transforms the linear predictors to the scale of the outcome variable (predicted probabilities). Finally, we can find the average predicted level.
+
+Combining these, we have:
+\[
+\hat{y}_{\tau = t} = \frac{1}{n} \sum_{i = 1}^n (1 + \exp(-\beta x_{i, \tau = t}))^{-1}
+\]
+We'll suppress the $\tau = t$ subscripts from here on.
+
+Now let's do this math in R: set treatment to 0 or 1, generate linear predictors, transform them to predicted outcomes, and take the mean.
+
+```{r}
+# Extract data from the model
+new <- lg$model
+# Set treatment to 0 for all observations
+new$treatment <- 0
+# Get linear predictors
+pred_ctl <- predict(lg, newdata = new)
+# Apply the inverse link function and take the mean
+mean_pred_ctl <- mean(plogis(pred_ctl))
+
+# Once again, with treatment
+new <- lg$model
+# Set treatment to 1 for all observations
+new$treatment <- 1
+# Get linear predictors
+pred_treat <- predict(lg, newdata = new)
+# Apply the inverse link function and take the mean
+mean_pred_treat <- mean(plogis(pred_treat))
+
+print(mean_pred_ctl)
+print(mean_pred_treat)
+```
+
+## Variance
+
+So far this is pretty straightforward, but we want to know the standard error of these estimates. As explained above, we can approximate their variance: if the gradient $\nabla {\hat{y}_{\tau = t}}$ is defined as $\mathbf{J}$, then
+
+\[\text{Var}(\hat{y}_{\tau = t}) =  \mathbf{J} \times \mathbf{V} \times \mathbf{J}^\prime\]
```

Comment:
@akgold tried to strip out most of the redundancy here and just refer to your section above - what do you think?